### ISSUE TITLE
Component: ConfirmDialog

### ISSUE THREAD

**Issue by shaman-apprentice** [primefaces/primeng#16861]
### Describe the bug

Confirmation Dialog has neither a confirm nor a cancle button.

### Environment

All

### Reproducer

Bug is present in current demo https://v18.primeng.org/confirmdialog

### Angular version

18.x.x

### PrimeNG version

18.0.0-rc.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

22.10.0

### Browser(s)

Chrome

### Steps to reproduce the behavior

1. Go to https://v18.primeng.org/confirmdialog
2. Click on "Delete"-button of first demo
3. Neither a confirm nor a cancle button is present.

### Expected behavior

Confirm and cancle button are present.

**Comment by ImClieo** [comment]
This seems to be due to some infinite loop happening when trying to determine a focus.
The specific error is a maximum call stack exceeded:
```
chunk-XM76ZT77.js:7 ERROR RangeError: Maximum call stack size exceeded
    at RegExp.exec (<anonymous>)
    at t.parseDurationToMilliseconds (chunk-TIN5GBLU.js:152:10562)
    at t.focus (chunk-TIN5GBLU.js:152:10718)
    at t.focus (chunk-TIN5GBLU.js:152:11018)
    at t.focus (chunk-TIN5GBLU.js:152:11018)
    at t.focus (chunk-TIN5GBLU.js:152:11018)
    at t.focus (chunk-TIN5GBLU.js:152:11018)
    at t.focus (chunk-TIN5GBLU.js:152:11018)
    at t.focus (chunk-TIN5GBLU.js:152:11018)
    at t.focus (chunk-TIN5GBLU.js:152:11018)
```

The dynamic dialog have the same kind of problem, this can however be fixed by setting the `focusOnShow` config option.

**Comment by shaman-apprentice** [comment]
I can confirm, that this was fixed with [18.0.0-rc.2](https://github.com/primefaces/primeng/milestone/297) - although it is not in the changelog.
